js/gameboard.js

```diff
@@ -1,110 +1,160 @@
+      var value = str[1]
+      searchObject[name] = strip(value, "%22")
+    }
+    return searchObject
+  }
+  return undefined
 }
-// just turn it blue
-function setBlue(){
-  color = "#0000ff";
-}
-// just turn it yellow
-function setYellow(){
-  color = "#ffff00";
+
+/**
+ * This gets the specfic search param in the href
+ */
+window.location.getSearchParam = function(name) {
+  return this.getSearchParams()[name]
 }
-// x-axis and y-axis; draws circle at that position
-function drawCircle(x, y){
-  // if x & y doesnt exist, get out of the function
-  if (! x || ! y){
-    console.log("Don't know where to go.");
-    return false;
-  } 
-  // canvas into the variable target
-  var target = getCanvas();
-  // assinging 2d to context 
-  var context = target.getContext("2d");
-  // shape that your drawing
-  context.beginPath();
-  var radius = 5; // the size is the radius in pixels    
-  var start = 0; 
-  var end = Math.PI*2;
-  context.arc(x, y, radius, start, end);
-  //calling the varible to the colors
-  context.fillStyle = color;
-  // filling the circles 
-  context.fill();
+
+/**
+ * This is the canvase class that is just a wrapper/refactor of the previous code base.
+ */
+function canvas() {
+  // the canvas element on the web page
+  this.canvasContext = $('#gameboard').get(0).getContext("2d")
+  
+  // the canvas's element offset
+  this._canvasOffet = $('#gameboard').offset();
+  
+  // the image being given by google MAP API in the href
+  this.imageUrl = strip(location.search.slice(1).split('screenshot')[1], '%22').slice(1)
+
+  // the color the pain brush is using
+  this.color = '';
+  
+  // some url
+  this.brickwall = "https://s-media-cache-ak0.pinimg.com/736x/94/aa/66/94aa6684bdc152457d6983e58fd24bd2.jpg";
+  
+  /**
+   * This renders the image into the canvas and sets the
+   * hieght and with of the image.
+   */
+  this.imageInput = function() {
+    // new image
+    var img = new Image();
+    var context = this.canvasContext;
+
+    // upon image loading 
+    img.onload = function () {
+      // positioning and draw the image 
+      context.drawImage(img, 0, 0);
+    };
+    // setting the empty image to an image 
+    img.src = this.imageUrl;
+    //prevents the image from stretching out 
+    img.width = "640px";
+    img.height = "480px";
+  }
+  
+  /**
+   * This chnages the color of the paint brush
+   * @param {string} color The color choosen
+   */
+  this.setColor = function(color) {
+    switch (color) {
+      case 'green':
+        this.color = "#009933";
+        break;
+      case 'red':
+        this.color = "#ff0000";
+        break;
+      case 'blue':
+        this.color = "#0000ff";
+        break;
+      case 'yellow':
+        this.color = "#ffff00";
+        break;
+      default:
+        if (!color) {
+          throw new Error('color given was undefined.')
+          return;
+        } else {
+          console.log('color does not exist')
+          return;
+        }
```

Comment:
fix spacing

Reply:
@andreGarvin depending on the editor you use, you can highlight all the text and use the reindent tool to fix any formatting.

Reply:
Depending on the editor you can highlight all of the text and use the reindent tool to reformat the code neatly.